clusterloader2/pkg/measurement/manager.go

```diff
@@ -42,14 +43,15 @@ func CreateMeasurementManager(clientSet clientset.Interface) *MeasurementManager
 }
 
 // Execute executes measurement based on provided identifier, methodName and params.
-func (mm *MeasurementManager) Execute(methodName string, identifier string, params map[string]interface{}) error {
+func (mm *MeasurementManager) Execute(methodName string, identifier string, params map[string]interface{}, clusterConfig *config.ClusterConfig) error {
 	measurementInstance, err := mm.getMeasurementInstance(methodName, identifier)
 	if err != nil {
 		return err
 	}
 	config := &MeasurementConfig{
-		ClientSet: mm.clientSet,
-		Params:    params,
+		ClientSet:     mm.clientSet,
+		Params:        params,
+		ClusterConfig: clusterConfig,
```

Comment:
Do you really need to pass and copy the cluster config each time for every measurement? It seems to me like the whole thing is just a global constant.

Reply:
I'm not convinced that global constant is the best solution. Here are the reasons why I think we shouldn't try this approach:
- Creates unnecessary dependency
It make metrics only be usable with this specific global variable. If anyone wants to use it in different project, he will need to assure that this global variable is initialized. Setting some external variable to change function result is not very intuitive. It also makes testing more complex.
- Scatters information across the whole project.
I assume that metrics is the part of project that at some point will be developed by other contributors from different projects. If we provide all available information in the interface signature, it's easier to understand what parameters are available and which not. IMHO going through the code to find the global variable you can use is more troublesome then just looking at the signature.

I think that we can make client config to be part of measurement manager (the same why as the clientSet is). This will remove cluster config from the signature of `Execute` of measurement manager.

Reply:
> It make metrics only be usable with this specific global variable. If anyone wants to use it in different project, he will need to assure that this global variable is initialized. 

If that's your problem, then just add another file in measurement pkg where you can initialize one single ClusterConfig object by calling some method at the start and then use the same object as a global constant for all measurements. Otherwise the problems with current style is:

- You're creating multiple copies of same global constant
- Storing constants as fields inside the struct
- Adding extra code complexity (due to new field that needs to be maintained, making function signatures more complex, asking measurement implementers to pass an extra arg in the Execute() method that they may not even need or have no clue about)

> Setting some external variable to change function result is not very intuitive.

I think this is the concept of global variables, and we do use them :)

> It also makes testing more complex.

It's not possible to write unit tests for these measurements even now. You're using methods like `SSH` which can't be mocked in the tests as the function is not passed as an argument to the measurement method - which should be the case in ideal unit-testing world where everything used in function should be passed as args to the function.

> If we provide all available information in the interface signature, it's easier to understand what parameters are available and which not.

I don't really agree with that we should pass all configurable stuff as params and not as constants. I believe in practice most people would just look at other measurement functions before implementing and it shouldn't be very non-trivial.

Reply:
I would say, that ClusterConfig should be constant.
And if we agree on that, this should be a constant field for MeasurementManager 9and then you don't need to pass it here).

Reply:
I added the ClusterConfig variable to ```measurement/manger.go```.